### ISSUE TITLE
Cuda 12: error: identifier "CUSPARSE_MM_ALG_DEFAULT" is undefined

### ISSUE THREAD

**Issue by romintomasetti** [kokkos/kokkos-kernels#1630]
I cannot compile Trilinos with CUDA 12.

It seems that this is caused by
```c++
src/impl/tpls/KokkosSparse_spmv_mv_tpl_spec_decl.hpp(182): error: identifier "CUSPARSE_MM_ALG_DEFAULT" is undefined
```
.

**Comment by ndellingwood** [comment]
Thanks for reporting @romintomasetti , I put in a PR #1632 to begin addressing some of the cuda/12 issues. One of our CI machines is currently down but hopefully a fix will make it into the repo some time next week

**Comment by ndellingwood** [comment]
Fixup merged to develop branch #1632

**Comment by ndellingwood** [comment]
@romintomasetti this update is now merged into Trilinos, thanks for reporting
